src/App.tsx

```diff
@@ -1,24 +1,195 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
+
 import { UserWarning } from './UserWarning';
+import { Header } from './components/Header/Header';
+import { TodoList } from './components/TodoList/TodoList';
+import { Footer } from './components/Footer/Footer';
+import { Notification } from './components/Notification/Notification';
+
+import { Todo } from './types/Todo';
+
+import {
+  getTodos, addTodo, onDelete, onUpdate,
+} from './api/todos';
 
-const USER_ID = 0;
+const USER_ID = 6476;
+
+export enum SortType {
+  ALL = 'All',
+  ACTIVE = 'Active',
+  COMPLETED = 'Completed',
+}
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterBy, setFilterBy] = useState('All');
+  const [isError, setIsError] = useState(false);
+  const [inputDisable, setInputDisable] = useState(false);
+  const [title, setTitle] = useState('');
+
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const visibleTodos = useMemo(() => {
+    return (todos.filter((todo) => {
+      switch (filterBy) {
+        case SortType.ALL:
+          return todos;
```

Comment:
you are not actually returning an array here. the filter callback should return a boolean value, so todos will be automatically transformed to a boolean (always `true`, because it's an array)
so you can join this case together with `default` case and return `true` for both